src/operators.jl

```diff
@@ -183,8 +185,55 @@ end
 function *{T<:Real}(g::DiGraph, v::Vector{T})
     length(v) == nv(g) || error("Vector size must equal number of vertices")
     y = zeros(T, nv(g))
-    for (i,j) in edges(g)
+    for p in edges(g)
+        i = p.first
```

Comment:
Let's use `src(p)` and `dst(p)` instead of `p.first` and `p.second` - `src()` and `dst()` will always work regardless of the underlying data structure of `Edge`.